pkgs/development/tools/sumneko-lua-language-server/default.nix

```
@@ -0,0 +1,47 @@
+{ stdenv, fetchFromGitHub, ninja, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  name = "sumneko-lua-language-server";
+  version = "1.9.0";
+
+  src = fetchFromGitHub {
+    owner = "sumneko";
+    repo = "lua-language-server";
+    rev = "${version}";
+    sha256 = "0d93ikxs0c5fm9k57xbzgphm90dnfqq3nz2bc3bqxi7gzyldb9z6";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    ninja
+    makeWrapper
+  ];
+
+  patches = [
+    ./sumneko_home_expansion.patch
+  ];
+
+  buildPhase = ''
+    cd 3rd/luamake
+    ninja -f ninja/linux.ninja
+    cd ../..
+    ./3rd/luamake/luamake rebuild
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/extras
+    cp -r ./{locale,meta,script,*.lua} $out/extras/
+    cp ./bin/Linux/{bee.so,lpeglabel.so} $out/extras
+    cp ./bin/Linux/lua-language-server $out/extras/.lua-language-server-unwrapped
+    makeWrapper $out/extras/.lua-language-server-unwrapped $out/bin/lua-language-server --add-flags "-E $out/extras/main.lua --logpath='~/.cache/sumneko_lua/log' --metapath='~/.cache/sumneko_lua/meta'"
```

```suggestion
    makeWrapper $out/extras/.lua-language-server-unwrapped \
      $out/bin/lua-language-server \
      --add-flags "-E $out/extras/main.lua \
      --logpath='~/.cache/sumneko_lua/log' \
      --metapath='~/.cache/sumneko_lua/meta'"
```

Mostly for better diffs.